feat(audit-logs): Adds plugin of OpenTelemetry Audit Logs (#684)
cloudoperators/greenhouse-extensions: audit-logs/README.md

- A log data-sink to store the audit-logs (e.g. OpenSearch). If you don't have one, reach out to your observability team to get access to one.
- We recommend a running cert-manager in the cluster before installing the Audit Logs Plugin
- To gather metrics from the audit logs collectors, you **must** install the [kube-monitoring](https://cloudoperators.github.io/greenhouse/docs/reference/catalog/kube-monitoring) Plugin first; it will install the necessary CRDs for, and install an instance of, the prometheus-operator.

**Step 1:**

You can install the `Audit Logs` plugin in your cluster with [Helm](https://helm.sh/docs/helm/helm_install) manually. 

Otherwise you can let the Greenhouse platform lifecycle do it for you automatically:
  1. Go to Greenhouse dashboard and select the **Audit Logs Plugin** from the catalog.
  1. Specify the cluster.
  1. Set the required option values.
  1. Create and specify a `Plugin` resource in your Greenhouse central cluster according to the [examples](#examples).

**Step 2:**

> The package will deploy the OpenTelemetry Operator which works as a manager for the collectors and auto-instrumentation of the workload. By default, the package will include a configuration for collecting openTelemetry. The log-collector is currently processing data from the [preconfigured receivers](#Overview): 

Reviewer: ```suggestion
The package will deploy the OpenTelemetry Operator which works as a manager for the collectors and auto-instrumentation of the workload. By default, the package will include a configuration for collecting audit-logs. The log-collector is currently processing data from the [preconfigured receivers](#Overview): 
```

- For logs, a OpenSearch instance to store. If you don't have one, reach out to your observability team to get access to one.
- We recommend a running cert-manager in the cluster before installing the OpenTelemetry Plugin
- To gather metrics, you **must** have a Prometheus instance in the onboarded cluster for storage and for managing Prometheus specific CRDs. If you don not have an instance, install the [kube-monitoring](https://cloudoperators.github.io/greenhouse/docs/reference/catalog/kube-monitoring) Plugin first. 

**Step 1:**

You can install the `OpenTelemetry` package in your cluster by installing it with [Helm](https://helm.sh/docs/helm/helm_install) manually or let the Greenhouse platform lifecycle do it for you automatically. For the latter, you can either:
  1. Go to Greenhouse dashboard and select the **OpenTelemetry** Plugin from the catalog. Specify the cluster and required option values.
  2. Create and specify a `Plugin` resource in your Greenhouse central cluster according to the [examples](#examples).

**Step 2:**

> The package will deploy the OpenTelemetry Operator which works as a manager for the collectors and auto-instrumentation of the workload. By default, the package will include a configuration for collecting metrics and logs. The log-collector is currently processing data from the [preconfigured receivers](#Overview): 

Reviewer: ```suggestion
The package will deploy the OpenTelemetry Operator which works as a manager for the audit-log collectors and auto-instrumentation of the workload. By default, the package will include a configuration for collecting audit-logs. The log-collector is currently processing data from the [preconfigured receivers](#Overview): 
```

- Files via the Filelog Receiver
- Kubernetes Events from the Kubernetes API server
- Journald events from systemd journal
- its own metrics

> You can disable the collection of logs by setting `openTelemetry.logCollector.enabled` to `false`. The same is true for disabling the collection of metrics by setting `openTelemetry.metricsCollector.enabled` to `false`.
> The `logsCollector` comes with a standard set of log-processing, such as adding cluster information and common labels for Journald events.
> In addition we provide default pipelines for common log types. Currently the following log types have default configurations that can be enabled (requires `logsCollector.enabled` to `true`):
>   1. KVM: `openTelemetry.logsCollector.kvmConfig`: Logs from Kernel-based Virtual Machines (KVMs) providing insights into virtualization activities, resource usage, and system performance
>   2. Ceph:`openTelemetry.logsCollector.cephConfig`: Logs from Ceph storage systems, capturing information about cluster operations, performance metrics, and health status

Reviewer: I don't know how it's intended but I would suggest removing ceph and kvm for now and focus on Kubernetes audit logs only. Like so:
```suggestion
You can disable the collection of audit-logs by setting `openTelemetry.auditLogCollector.enabled` to `false`. 
```

Note: `openTelemetry.auditLogCollector` does not currently exist.

> These default configurations provide common labels and Grok parsing for logs emitted through the respective services.
> 
> Based on the backend selection the telemetry data will be exporter to the backend.

Reviewer: Is that true still?
```suggestion
These default configurations provide common labels and Grok parsing for logs emitted through the respective services.

Based on the backend selection the telemetry data will be exporter to the backend.
```

**Step 3:**

> Greenhouse regularly performs integration tests that are bundled with **OpenTelemetry**. These provide feedback on whether all the necessary resources are installed and continuously up and running. You will find messages about this in the plugin status and also in the Greenhouse dashboard. 

Reviewer: ```suggestion
Greenhouse regularly performs integration tests that are bundled with **Audit-Logs**. These provide feedback on whether all the necessary resources are installed and continuously up and running. You will find messages about this in the plugin status and also in the Greenhouse dashboard. 
```

> ## Failover Connector
> 
> The **OpenTelemetry** Plugin comes with a [Failover Connector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/failoverconnector) for OpenSearch for two users. The connector will periodically try to establish a stable connection for the prefered user (`failover_username_a`) and in case of a failed try, the connector will try to establish a connection with the fallback user (`failover_username_b`). This feature can be used to secure the shipping of logs in case of expiring credentials or password rotation.

Reviewer: Please check with team if this applies to audit-logs as well. 
```suggestion
## Failover Connector

The **Audit-Logs** Plugin comes with a [Failover Connector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/failoverconnector) for OpenSearch for two users. The connector will periodically try to establish a stable connection for the prefered user (`failover_username_a`) and in case of a failed try, the connector will try to establish a connection with the fallback user (`failover_username_b`). This feature can be used to secure the shipping of logs in case of expiring credentials or password rotation.
```

`openTelemetry.cluster`                  | Cluster label for logging        | string | `false` |
`openTelemetry.prometheus.additionalLabels`               | Label selector for Prometheus resources to be picked-up by the operator | map | `false` | 
`openTelemetry.prometheus.rules.additionalRuleLabels`               | Additional labels for PrometheusRule alerts | map | `false` | 
`openTelemetry.prometheus.serviceMonitor.enabled` | Activates the service-monitoring for the Logs Collector  | bool | `false` | 
`openTelemetry.prometheus.podMonitor.enabled`       | Activates the pod-monitoring for the Logs Collector | bool | `false` | 
`openTelemetry.prometheus.rules.create`       | Enables PrometheusRule resources to be created | bool | `false` | 
`openTelemetry.prometheus.rules.disabled`       | List of PrometheusRules to disable | map | `false` | 
`openTelemetry.prometheus.rules.labels`       | Labels for PrometheusRules | map | `false` | 
`openTelemetry.prometheus.rules.annotations`       | Annotations for PrometheusRules | map | `false` | 
`openTelemetry.prometheus.rules.additionalRuleLabels`       | Additional labels for PrometheusRule alerts,  | map | `false` | 

> `opentelemetry-operator.admissionWebhooks.certManager.enabled` | Activate to use the CertManager for generating self-signed certificates | bool | `false` | 
> `opentelemetry-operator.admissionWebhooks.autoGenerateCert.enabled` | Activate to use Helm to create self-signed certificates | bool | `false` | 
> `opentelemetry-operator.admissionWebhooks.autoGenerateCert.recreate` | Activate to recreate the cert after a defined period (certPeriodDays default is 365) | bool | `false` | 
> `opentelemetry-operator.kubeRBACProxy.enabled` | Activate to enable Kube-RBAC-Proxy for OpenTelemetry | bool | `false` | 
> `opentelemetry-operator.manager.prometheusRule.defaultRules.enabled` | Activate to enable default rules for monitoring the OpenTelemetry Manager | bool | `false` | 
> `opentelemetry-operator.manager.prometheusRule.enabled` | Activate to enable rules for monitoring the OpenTelemetry Manager | bool | `false` | 

Reviewer: ```suggestion
`audit-logs-operator.admissionWebhooks.certManager.enabled` | Activate to use the CertManager for generating self-signed certificates | bool | `false` | 
`audit-logs-operator.admissionWebhooks.autoGenerateCert.enabled` | Activate to use Helm to create self-signed certificates | bool | `false` | 
`opentelemetry-operator.admissionWebhooks.autoGenerateCert.recreate` | Activate to recreate the cert after a defined period (certPeriodDays default is 365) | bool | `false` | 
`audit-logs-operator.kubeRBACProxy.enabled` | Activate to enable Kube-RBAC-Proxy for OpenTelemetry | bool | `false` | 
`audit-logs-operator.manager.prometheusRule.defaultRules.enabled` | Activate to enable default rules for monitoring the OpenTelemetry Manager | bool | `false` | 
`audit-logs-operator.manager.prometheusRule.enabled` | Activate to enable rules for monitoring the OpenTelemetry Manager | bool | `false` | 
```
